packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -0,0 +1,294 @@
+/**
+  @module ember-data
+*/
+
+import Serializer from "ember-data/system/serializer";
+
+var copy = Ember.copy;
+var get = Ember.get;
+var forEach = Ember.ArrayPolyfills.forEach;
+var map = Ember.ArrayPolyfills.map;
+
+/**
+  @class JSONAPISerializer
+  @namespace DS
+  @extends DS.Serializer
+*/
+export default Serializer.extend({
+
+  keyForAttribute: function(key) {
+    return Ember.String.dasherize(key);
+  },
+
+  keyForRelationship: function(key, kind) {
+    return Ember.String.dasherize(key);
```

Comment:
I believe JSONAPI dropped the recommendation that these keys be dasherized?

For both `keyForAttribute` and `keyForRelationship` I think we should no-op these methods and allow users to override them to best suit their api.